### PR TITLE
BrowserStack.Sangavi

### PR DESCRIPTION
**Pull request by sangavim**

### CODE REVIEW

File: BrowserStack.java [hunk 1]
```diff
@@ -34,36 +34,83 @@ public static void myassert(boolean  x) {
...
     private  void push(String url){
-        //WRITE YOUR CODE HERE
+        top++;
+        myStack[top] = url;
     }
+
     //pop()- pops the element from the stack
     private String pop(){
-        //WRITE YOUR CODE HERE
+        String popElement = myStack[top];
```
Comment (balamuralim) [inline BrowserStack.java:46]: Memory leak issue. Though you decrement the pointer. The old value still resides in the stack until you override with a new value.

`` myStack[top] = null; ``
will impress the interviewer.

File: BrowserStack.java [hunk 2]
```diff
@@ -34,36 +34,83 @@ public static void myassert(boolean  x) {
...
+        }
+        return empty;
     }
-    private  boolean search(String url){
-        //WRITE YOUR CODE HERE
 
+    //Search an element 
+    private boolean search(String url){
+        int i = 0, j = 0;
+        String[] tempStack = new String[16];
```
Comment (sarankris) [inline BrowserStack.java:91]: If you are using a new stack to search, then it sufficient to initialze with the size of 'top+1'.

File: BrowserStack.java [hunk 3]
```diff
@@ -34,36 +34,83 @@ public static void myassert(boolean  x) {
             throw new IllegalArgumentException("Assert fail") ;
         }
     }
+    
     //push() - pushes the element in this case its a url (String)
     private  void push(String url){
-        //WRITE YOUR CODE HERE
+        top++;
```
Comment (sarankris) [inline BrowserStack.java:40]: Stack was initialized to size of 16. So, we should check for Stack full condition and return appropriate error.

File: BrowserStack.java (hunk 2 above)
Comment (balamuralim) [reply BrowserStack.java:91]: Try to avoid constants as much possible in initializing temp arrays

File: BrowserStack.java [hunk 4]
```diff
@@ -34,36 +34,83 @@ public static void myassert(boolean  x) {
...
+            } else {
+                tempStack[i] = pop();
+                i++;
+            }
+        }
+        if (!found){
+            System.out.println(url+" is not in stack");
+        }
+        for (j = i-1; j >= 0; j--){
+            top++;
```
Comment (balamuralim) [inline BrowserStack.java:108]: **Reusability** : push(tempStack[j]) method can be used instead

File: BrowserStack.java [hunk 5]
```diff
@@ -34,36 +34,83 @@ public static void myassert(boolean  x) {
...
+    
+    //check whether the stack is Empty
     private  boolean isEmpty(){
-        //WRITE YOUR CODE HERE
-
+        int stackSize = size();
+        boolean empty = false;
+        if(stackSize == 0){
+            System.out.println("Empty Stack");
+            empty = true;
```
Comment (balamuralim) [inline BrowserStack.java:80]: **Minor** : can return directly _true_ from here.

**Comment by balamuralim** [comment]
Nice effort @sangavim. Keep it up 👍

**Pull request closed without merging**